### DISCUSSION TITLE
Scaling photoprism horizontally on Kubernetes

### DISCUSSION THREAD

**Discussion by srgvg** [photoprism/photoprism#1212] [Q&A] [answered]
New to photoprism, I recently installed it - manually, not with the helm chart - as a deployment with a single replica on Kubernetes. Now I'm wondering if photoprism can scale horiizontally, hence increasing the numer of replica pods - to make use of more cpu and memory by running it on more than one node.

I assume at the very least, this needs
- using a mysql db, not sqlite
- using shared RWX  persistent volume for the storage (which is shared amongst pods in a deployement, whereas a each pod in a statefulset gets it's own dedicated volume ***)

Other than that, I'm not sure how and if scaling to multiple pods whould work out?

---
*** which makes me wonder why the official helm chart uses a statefull set. Whilst in the values it doesn not advertise the parameter to scale the replica's, it does have a variable for this in the template. I'm not sure how that could work, having distinct pods that each work on it's own storage?

Also: thanks for sharing this wonderful piece of software, which allows for different setup and workflows, and doesn't force itself in the way!

**Comment by kvalev** [comment, marked as answer]
While the scaling itself would work, I dont think it makes much sense. For the scaling to be more meaningful you will have to decouple the frontend and backend, but most importantly you will have to introduce a distributed indexer, as this is the only part worth scaling imo.

**Reply by lastzero** [reply]
Yep, lots of work. We'll get to this eventually, but right now we're focusing on home users who don't need massive scalability like SaaS providers who host it for others.
